source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -460,7 +471,7 @@ ComponentInterface<NodeT>::on_set_parameters_callback(const std::vector<rclcpp::
       auto new_parameter = modulo_core::translators::read_parameter_const(ros_parameter, parameter);
```

Comment:
This is problematic when `parameter` is empty. Because the ros_parameter is of type string when you add_parameter, new_parameter will _not_ be empty. This means that `validate_parameter` will see a different, non-empty, default-valued version of the parameter. For an integer, this might mean that we want to add a "not set" number, but validate parameter will see a value of `0` instead.